vulcan/lib/orchestration.rb

```diff
@@ -282,73 +269,77 @@ def build_target_for(step_name, cache = {})
-            script[input.id] = input_files.last
-          end
-        elsif step_name == :primary_outputs
-          script = {}
-          workflow.outputs.each do |output|
-            output_filenames << output.id
-
-            source_step_name, source_output_name = output.outputSource
-            input_file = build_target_for(source_step_name, cache).take_as_input(source_output_name, output.id)
-            script[output.id] = input_file
-            if input_file.nil?
-              raise "Could not find output #{source_output_name.inspect} from step #{source_step_name.inspect} for primary output #{output.id.inspect}"
-            end
-            input_files << input_file
-          end
-        elsif (step = workflow.find_step(step_name))
-          if step.ui_behavior?
+          output_filenames = []
+          input_files = []
+          script = nil
+
+          if step_name.is_a?(Symbol) && step_name =~ /^primary_inputs_/
+            var_name = step_name.to_s.sub(/^primary_inputs_/, "")
             script = {}
-          elsif step.script_name
-            script = step.lookup_operation_script
-            raise "Could not find backing script #{step.script_name.inspect} for step #{step.id}" if script.nil?
-          else
-            raise "Step #{step.id} has invalid run: #{step.run}.  Must be either a ui-queries/, ui-outputs/, or scripts/ entry." if script.nil?
-          end
+            workflow.inputs.zip(primary_input_material_sources).each do |input, source|
+              next if var_name != input.id
 
-          step.out.each do |step_out|
-            output_filenames << step_out.id
+              output_filenames << input.id
+              input_files << source.take_as_input(input.id)
+              script[input.id] = input_files.last
+            end
+          elsif step_name == :primary_outputs
+            script = {}
+            workflow.outputs.each do |output|
+              output_filenames << output.id
+
+              source_step_name, source_output_name = output.outputSource
+              input_file = build_target_for(source_step_name, cache).take_as_input(source_output_name, output.id)
+              script[output.id] = input_file
+              if input_file.nil?
+                raise "Could not find output #{source_output_name.inspect} from step #{source_step_name.inspect} for primary output #{output.id.inspect}"
+              end
+              input_files << input_file
+            end
+          elsif (step = workflow.find_step(step_name))
             if step.ui_behavior?
-              ref = session.material_reference_for([step_name, step_out.id])
-              source = material_source(ref)
-              input_files << source.take_as_input(step_out.id)
-              script[step_out.id] = input_files.last
+              script = {}
+            elsif step.script_name
+              # Check snapshot first for the stored script
+              script = session.snapshot_script(step.id)
```

Comment:
In the midst of the formatting change, this is the change that first checks for the script being defined in the snapshot. If it's not there, then take the current definition on disk. This seemed better than pinging GH for a raw script file, since we'd have to store the path to the file anyways in order to fetch it.